aio/aio-proxy/aio_proxy/search/es_search_runner.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from datetime import timedelta
+
+from aio_proxy.search.cache.cache import cache_strategy
+from aio_proxy.search.es_index import ElasticsearchSireneIndex
+from aio_proxy.search.geo_search import geo_search_builder
+from aio_proxy.search.text_search import text_search_builder
+
+TIME_TO_LIVE = timedelta(days=31)
+MIN_EXECUTION_TIME = 400
+MAX_TOTAL_RESULTS = 10000
+
+
+class ElasticSearchRunner:
+    def __init__(self, search_params, search_type="text"):
+        self.es_search_client = ElasticsearchSireneIndex.search()
+        self.search_type = search_type
+        self.search_params = search_params
+        self.is_text_search = False
+        self.es_search_results = None
+        self.total_results = None
+        self.execution_time = None
+        self.run()
+
+    def build_es_search_text_query(self):
+        text_search_builder(self)
+
+    def build_es_search_geo_query(self):
+        geo_search_builder(self)
+
+    def sort_es_search_query(self):
+        # Sorting is very heavy on performance if there are no
+        # search terms (only filters). As there is no search terms, we can
+        # exclude this sorting because score is the same for all results
+        # documents. Beware, nom and prenoms are search fields.
+        if self.is_text_search:
+            self.es_search_client = self.es_search_client.sort(
+                {"_score": {"order": "desc"}},
+                {"etat_administratif_unite_legale": {"order": "asc"}},
+            )
+        # If only filters are used, use nombre établissements ouverts to sort the
+        # results
+        else:
+            self.es_search_client = self.es_search_client.sort(
+                {"nombre_etablissements_ouverts": {"order": "desc"}},
+            )
+
+    def execute_and_format_es_search(self):
+        offset = self.search_params.page
```

Comment:
Actually, `offset` is not an elastic search term, the documentation uses `from` but since it's a keyword in python I cannot use it. And `offset` seems to me like a good alternative.
What do you think ? Do you have a better var name ?